File: cDevelop/Forms/frmProductoList.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Data.SqlClient;

namespace cDevelop.Forms
{
    public partial class frmProductoList : dcLibrary.dcParentList
    {
        public frmProductoList()
        {
            InitializeComponent();
        }
        public override void Init()
        {
            SQLGridParm = new SqlParameter();
            SQLGridParm.ParameterName = "@ProductoID";        
            SQLGridParm.Value = 0;

            SqlDataAdapter adp = new SqlDataAdapter();
            adp.SelectCommand = dcGral.getSQLCommand(Connect, "spProductoSelect", SQLGridParm);
            adp.InsertCommand = dcGral.getSQLCommand(Connect, "spProductoInsert");
            adp.UpdateCommand = dcGral.getSQLCommand(Connect, "spProductoUpdate");
            adp.DeleteCommand = dcGral.getSQLCommand(Connect, "spProductoDelete");
            adp.InsertCommand.Parameters[0].Direction = ParameterDirection.InputOutput;
            Adaptador = adp;
            base.Init();
        }
        private void frmProductoList_Load(object sender, EventArgs e)
        {

        }
    }
}

File: vbDevelop/Forms/frmProductoList.vb
Imports System.Data.SqlClient
Public Class frmProductoList

    Public Overrides Sub Init()
        SQLGridParm = New SqlParameter("@ProductoID", 0)

        Dim adp As New SqlDataAdapter

        adp.SelectCommand = dcGral.getSQLCommand(Connect, "spProductoSelect", SQLGridParm)
        adp.InsertCommand = dcGral.getSQLCommand(Connect, "spProductoInsert")
        adp.UpdateCommand = dcGral.getSQLCommand(Connect, "spProductoUpdate")
        adp.DeleteCommand = dcGral.getSQLCommand(Connect, "spProductoDelete")
        adp.InsertCommand.Parameters(0).Direction = ParameterDirection.InputOutput
        Adaptador = adp
        MyBase.Init()
    End Sub

    Private Sub frmProductoList_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub
End Class
